editoast/core_task/src/envs/pathfinding.rs

```diff
@@ -337,44 +278,127 @@ where
             .collect()
```

Comment:
Just because I bumped into this code, when reviewing... The `.cloned().cloned()` could probably be replaced with `.copied().cloned()`.